test/type.test.js

```diff
@@ -2,14 +2,19 @@ import test from 'ava';
 import sinon from 'sinon';
 import proxyquire from 'proxyquire';
 
-const spy = sinon.spy();
-const BemEntity = proxyquire('../index', {
-    'bem-naming': {
-        typeOf: spy
-    }
+test.beforeEach(t => {
+    t.context.spy = sinon.spy();
+    t.context.BemEntity = proxyquire('../index', {
+        'bem-naming': {
+            typeOf: t.context.spy
+        }
+    });
 });
 
+
 test('should use `naming.typeOf()` for block', t => {
+    const BemEntity = t.context.BemEntity;
+    const spy = t.context.spy;
     const entity = new BemEntity({ block: 'block' });
```

Comment:
I don't like the same code in different tests. May be use `t.BemEntity` and `t.spy`?

```js
const entity = new t.BemEntity({ block: 'block' });

t.ok(t.spy.calledWith({}))
```